.pipelines/templates/PackageTestResultsAnalysis.yml

```diff
@@ -0,0 +1,295 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+parameters:
+  - name: authenticateToPip
+    type: boolean
+    default: true
+
+  - name: buildRepoRoot
+    type: string
+    default: "$(Build.SourcesDirectory)"
+
+  - name: failOnTestFailures
+    type: boolean
+    default: true
+
+  - name: outputArtifactsFolder
+    type: string
+    default: "$(Build.ArtifactStagingDirectory)"
+
+  # Local constants. Can't use variables in a template without jobs or stages.
+  - name: reportFileName
+    type: string
+    default: "pkgtest_report_junit.xml"
+
+  - name: testsWorkspace
+    type: string
+    default: "$(Agent.TempDirectory)"
+
+steps:
+  - ${{ if parameters.authenticateToPip }}:
+      - task: PipAuthenticate@1
+        inputs:
+          artifactFeeds: "mariner/Mariner-Pypi-Feed"
+        displayName: "Authenticate to pip"
+
+  - bash: |
+      pip3 install junit_xml
+    displayName: "Install Python dependencies"
+
+  - task: PythonScript@0
+    inputs:
+      pythonInterpreter: "/bin/python3"
+      scriptSource: "inline"
+      script: |
```

Comment:
Should we probably have a file instead of a large script ?
We could then call that script within this

Reply:
I'm trying to avoid scripts inside all GitHub templates as they are made to be ran on `steps` level and thus cannot check-out their own repositories and access files in them. I agree, that I'd prefer to have a script in a separate file, but so far I couldn't find any good way to make it not require us to now clone the whole repo in the pipelines just to get access to the script file. Expanding pipeline templates doesn't require cloning the repo, thus inlining a script works.

One more idea that I had on my mind, would be to enclose this script inside its own separate pipeline template to make it more reusable in other pipelines. That still didn't make it possible to run the script manually, though, and added additional complexity.

Having said that, I am open to ideas! I'd love to learn if there's a better way.

Reply:
Ohh, I am not aware if there is a better way to do it.
It felt a little off, which is why the comment :)